cordova-lib/src/platforms/PlatformApi.js

```diff
@@ -0,0 +1,365 @@
+
+    var baseCfg = new ConfigParser(baseConfig);
+    var ownCfg = new ConfigParser(ownConfig);
+    util.mergeXml(baseCfg.doc.getroot(), ownCfg.doc.getroot(), this.platform, true);
+
+    // CB-6976 Windows Universal Apps. For smooth transition and to prevent mass api failures
+    // we allow using windows8 tag for new windows platform
+    // TODO: this section should be moved to windows platform's implementation of PlatformApi
+    if (this.platform == 'windows') {
+        util.mergeXml(baseCfg.doc.getroot(), ownCfg.doc.getroot(), 'windows8', true);
+    }
+
+    return Q.when(ownCfg.write());
+};
+
+/**
+ * Base implementation for getWwwDir. Assumes that
+ * www directory is placed at the root of project.
+ * @return {String} Path to platform's www directory.
+ */
+BasePlatformApi.prototype.getWwwDir = function() {
+    if (this.parser && this.parser.www_dir) {
+        return this.parser.www_dir();
+    }
+
+    return  path.join(this.root, 'www');
+};
+
+/**
+ * Base implementation for getCordovaJsSrc. Assumes that cordova.js
+ * source is placed at the root of platform's source dir.
+ * @return {String} Path to platform's 'cordova-js-src' folder.
+ */
+BasePlatformApi.prototype.getCordovaJsSrc = function(platformSource) {
+    if (this.parser && this.parser.cordovajs_src_path) {
+        return this.parser.cordovajs_src_path(platformSource);
+    }
+
+    return path.resolve(platformSource, 'cordova-js-src');
+};
+
+/**
+ * Base implementation for updateWww.
+ * @param {string} [wwwSource] Source dir for www files. If not provided, method
+ *                             will try to find www directory from cordova project
+ */
+BasePlatformApi.prototype.updateWww = function(wwwSource) {
+    if (this.parser && this.parser.update_www) {
+        return Q.when(this.parser.update_www());
+    }
+
+    if (!wwwSource) {
+        var projectRoot = util.isCordova(this.root);
+        wwwSource = util.projectWww(projectRoot);
+    }
+    var platformWww = path.join(this.root, 'platform_www');
+
+    // Clear the www dir
+    shell.rm('-rf', this.getWwwDir());
+    shell.mkdir(this.getWwwDir());
+    // Copy over all app www assets
+    shell.cp('-rf', path.join(wwwSource, '*'), this.getWwwDir());
+    // Copy over stock platform www assets (cordova.js)
+    runShellSilent(function () {
+        shell.cp('-rf', path.join(platformWww, '*'), this.getWwwDir());
+    }.bind(this));
+
+    return Q();
+};
+
+/**
+ * Base implementation for updateProject. Does nothing
+ * since implementation is heavily depends on platform specifics.
+ * Always should be overridden by platform.
+ */
+BasePlatformApi.prototype.updateProject = function(configSource) {
+    if (this.parser && this.parser.update_project) {
+        var configParser = new ConfigParser(configSource || this.getConfigXml());
+        return Q.when(this.parser.update_project(configParser));
+    }
+    return Q();
```

Comment:
Well, in some cases platform may want to do nothing in `updateProject`. Example is a browser platform, which needs to update config.xml  and www content only during prepare.